sktime/transformations/panel/supervised_intervals.py

```diff
@@ -0,0 +1,471 @@
+                Series: pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+                Panel: pd.DataFrame with 2-level MultiIndex, list of pd.DataFrame,
+                    nested pd.DataFrame, or pd.DataFrame in long/wide format
+                subject to sktime mtype format specifications, for further details see
+                    examples/AA_datatypes_and_datasets.ipynb
+        y : Series or Panel, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        type depends on type of X and scitype:transform-output tag:
+            |   `X`    | `tf-output`  |     type of return     |
+            |----------|--------------|------------------------|
+            | `Series` | `Primitives` | `pd.DataFrame` (1-row) |
+            | `Panel`  | `Primitives` | `pd.DataFrame`         |
+            | `Series` | `Series`     | `Series`               |
+            | `Panel`  | `Series`     | `Panel`                |
+            | `Series` | `Panel`      | `Panel`                |
+        instances in return correspond to instances in `X`
+        combinations not in the table are currently not supported
+
+        Explicitly, with examples:
+            if `X` is `Series` (e.g., `pd.DataFrame`) and `transform-output` is `Series`
+                then the return is a single `Series` of the same mtype
+                Example: detrending a single series
+            if `X` is `Panel` (e.g., `pd-multiindex`) and `transform-output` is `Series`
+                then the return is `Panel` with same number of instances as `X`
+                    (the transformer is applied to each input Series instance)
+                Example: all series in the panel are detrended individually
+            if `X` is `Series` or `Panel` and `transform-output` is `Primitives`
+                then the return is `pd.DataFrame` with as many rows as instances in `X`
+                Example: i-th row of the return has mean and variance of the i-th series
+            if `X` is `Series` and `transform-output` is `Panel`
+                then the return is a `Panel` object of type `pd-multiindex`
+                Example: i-th instance of the output is the i-th window running over `X`
+        """
+        self.reset()
+        if y is None:
+            raise ValueError("SupervisedIntervals requires `y` in `fit`.")
+        X, y = self._check_X_y(X=X, y=y)
+
+        y = self._fit_setup(X, y)
+        X_norm = z_normalise_series_3d(X)
+
+        fit = Parallel(n_jobs=self._threads_to_use)(
+            delayed(self._generate_intervals)(
+                X,
+                X_norm,
+                y,
+                i,
+                True,
+            )
+            for i in range(self.n_intervals)
+        )
+
+        (
+            intervals,
+            transformed_intervals,
+        ) = zip(*fit)
+
+        self._intervals = []
+        for i in intervals:
```

Comment:
`self._intervals = list(intervals)` ?

Reply:
this would create a list of lists rather than a single list im pretty sure